[ci] Add rule to not allow code after `then` in conditional

claywar opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Adds CI rule to check for any non-comment text after the word `then` in lua scripts and corrects all issues raised from it.
<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->

## Steps to test these changes
No changes should be observed
<!-- Clear and detailed steps to test your changes here -->

Xaver-DaRed commented on scripts/actions/mobskills/astral_flow_pet.lua:
```diff
@@ -42,18 +55,12 @@ mobskillObject.onMobWeaponSkill = function(target, mob, skill)
-        printf("[astral_flow_pet] received unexpected pet family %i. Defaulted skill to Searing Light.", petFamily)
+    for mobSkillId, petFamilyList in ipairs(petAstralFlowAbility) do
+        if utils.contains(petFamily, petFamilyList) then
+            skillId = mobSkillId
```
no break for the loop when a match happens?

Xaver-DaRed replied:
Also, `skillId` default value is already 919. Seems reduntant to set it to 919 again if a match happens.

claywar replied:
You're right, this is a carryover from a way it was previously handled

Xaver-DaRed reviewed (approved):
Nitpicks apart, lgtm

TeoTwawki commented on scripts/actions/spells/black/bio_iii.lua:
```diff
@@ -45,20 +45,31 @@ spellObject.onSpellCast = function(caster, target, spell)
     -- this is a tiered calculation that has at least three tiers,
     -- so I'll use breakpoints for human readability
-    local dotdmg = 0
-    if     skillLvl > 400 then dotdmg = 17
```
ya know, I even called out this exact styling in the style guide years ago. _Reeeeeeee no padding between the if and the condition_

Ty for fixing.

claywar committed:
[ci] Add rule to not allow code after `then` in conditional

Update unit tests for new rule

Remove debugging change

Add break and comment for astral flow pet
